src/ILCompiler/src/Program.cs

```diff
@@ -446,6 +456,17 @@ private int Run(string[] args)
             ObjectDumper dumper = _mapFileName != null ? new ObjectDumper(_mapFileName) : null;
 
             CompilationResults compilationResults = compilation.Compile(_outputFilePath, dumper);
+            if (_nativeLib)
+            {
+                DefFileWriter defFileWriter = new DefFileWriter(typeSystemContext, _exportsFile);
+                foreach (var compilationRoot in compilationRoots)
+                {
+                    if (compilationRoot is ExportedMethodsRootProvider)
```

Comment:
Nit: You can use new C# features to avoid the cast.

```
 if (compilationRoot is ExportedMethodsRootProvider provider)
     defFileWriter.AddExportedMethod(provider);
```